design/vela-core/environment.md

```diff
@@ -1,294 +1,262 @@
 
-- K8s Clusters. Different environments might have clusters of different size and version, e.g.
-  small v1.10 k8s cluster for *dev* environment and large v1.9 k8s cluster for *prod* environment.
-- Admin policies. Production environment would usually set global policies on all application deployments,
+- Kubernetes clusters. These include not only existing clusters but also new clusters to create during environment setup.
+- Admin policies. Production environment would usually set global policies for all application deployments,
   like chaos testing, SLO requirements, security scanning, misconfiguration detection.
-- Operators & CRDs. Environments contains a variety of Operators & CRDs as system capabilities.
-  These operators can include domain registration, routing, logging, monitoring, autoscaling, etc.
+- System components. That include installing system level Operators & CRDS like FluxCD, KEDA, OpenKruise,
+  shared Namespaces, Observability (Prometheus, Grafana, Loki), Terraform Controller, KubeFlow Controller, etc.
 - Shared services. Environments contains a variety of system services that are be shared by applications.
   These resources can include DB, cache, load balancers, API Gateways, etc.
 
-With the introduction of environment, the user workflow on Vela works like:
+In this doc, we will provide solutions and case studies on how to use KubeVela to make environments, initialize shared resources,
+and compose application rollout across multiple environments.
 
-- Platform team defines system Components for Environments.
-- Platform team initializes Environments using system Components.
-- Platform team defines user Components for Applications.
-- Developers choose user Components and Environments to deploy Applications.
 
-When initializing environments, there are additional requirements:
+## Using KubeVela to initialize Environment
 
-- The ability to describe dependency relations between environments. For example, both *dev* and *prod* environments
-  would depend on a common environment that sets up operators both *dev* and *prod* would use.
+An environment is just a logical concept that groups a bundle of resources for multiple applications to share.
+To make an environment, we first need to initialize the shared resources.
 
-In the following we propose solutions to implement environment concept for application delivery.
+When initializing an environment, there is one additional requirement:
 
-## Initializer CRD
+- The ability to describe dependencies between environment modules. For example, both *dev* and *prod* environments
+  would depend on a common module that installs operators both *dev* and *prod* would use.
 
-We propose to add an Initializer CRD:
+To setup shared resources and define dependencies, we propose to use Application with two Workflow Steps
+to achieve the requirements. An example looks like below:
 
 ```yaml
-kind: Initializer
+apiVersion: core.oam.dev/v1beta1
+kind: Application
```

Comment:
I am still curious when an application can be called an environment? I should have missed something important.

Reply:
We will focus on how to use KubeVela to support environment-based application rollout.
We don't call an Application an environment, but focus on how we use KubeVela to support various scenarios.